src/Browser.php

```diff
@@ -784,7 +880,55 @@ private function requestMayBeStreaming($method, $url, array $headers = array(),
         }
 
         return $this->transaction->send(
-            new Request($method, $url, $headers, $body, $this->protocolVersion)
+            new Request($method, $url, $this->mergeHeaders($headers), $body, $this->protocolVersion)
         );
     }
+
+    /**
+     * @param array $headers
+     * @return array
+     */
+    private function mergeHeaders($headers)
+    {
+        foreach (array_keys($this->defaultHeaders) as $key) {
```

Comment:
nit: all global functions should be prefixed with `\` as per #330 (see also following calls)

Reply:
added \ to the functions